### PR TITLE
Allow specifying safeFields #7

### PR DESCRIPTION
**Pull request by zbarbuto**
### Description
Allow specifying an array of safe field properties to be included in 4xx and 5xx errors #7

#### Related issues
- #7 Implement "safeFields" option

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by bajtos** [comment]
@slnode ok to test

File: lib/data-builder.js [hunk 1]
```diff
@@ -67,7 +71,7 @@ function fillDebugData(data, err) {
   cloneAllProperties(data, err);
 }
 
-function fillBadRequestError(data, err) {
+function fillBadRequestError(data, err, safeFields) {
```
Comment (bajtos) [inline lib/data-builder.js]: AFAICT, this new argument is not used at all, please revert.

File: test/handler.test.js [hunk 2]
```diff
@@ -231,6 +231,51 @@ describe('strong-error-handler', function() {
         });
       });
 
+    it('should allow setting safe fields when status=5xx', function(done) {
+      // Mock an error reported by fs.readFile
```
Comment (bajtos) [inline test/handler.test.js]: The comment is not accurate, the error object does not resemble errors reported by `fs.readFile`. Please remove this line.

File: test/handler.test.js [hunk 3]
```diff
@@ -231,6 +231,51 @@ describe('strong-error-handler', function() {
...
+        expect(res.body).to.have.property('error');
+        expect(res.body.error).to.have.property('safeField', 'SAFE');
+        expect(res.body.error).not.to.have.property('unsafeField');
+
+        done();
+      });
+    });
+
+    it('should allow setting safe fields when status=4xx', function(done) {
+      // Mock an error reported by fs.readFile
```
Comment (bajtos) [inline test/handler.test.js]: Ditto: The comment is not accurate, the error object does not resemble errors reported by `fs.readFile`. Please remove this line.

File: lib/handler.js [hunk 4]
```diff
@@ -49,7 +49,7 @@ exports = module.exports = function createStrongErrorHandler(options) {
     if (!err.status && !err.statusCode && res.statusCode >= 400)
       err.statusCode = res.statusCode;
 
-    var data = buildResponseData(err, isDebugMode);
+    var data = buildResponseData(err, options);
```
Comment (bajtos) [inline lib/handler.js]: With this change in place, the variable `isDebugMode` is no longer used and should be removed.

**Comment by zbarbuto** [comment]
@bajtos Thanks for feedback. Few mistakes from some copy-pastes. PR updated.

File: lib/handler.js (hunk 4 above)
Comment (zbarbuto) [inline lib/handler.js]: @bajtos Thanks for feedback. Few mistakes from some copy-pastes. PR updated.

**Comment by bajtos** [comment]
Landed, thank you for the contribution! 🎉

**Comment by bajtos** [comment]
@zbarbuto actually, I forgot to ask you to add an entry to README to document the new option, see https://github.com/strongloop/strong-error-handler/blob/master/README.md#options. Could you please add the docs and open a new pull request?

**Pull request merged**